cou/apps/app.py

```diff
@@ -335,19 +482,29 @@ def apt_source_codename(self) -> Optional[OpenStackRelease]:
     def channel_codename(self) -> OpenStackRelease:
         """Identify the OpenStack release set in the charm channel.
 
+        When application comes from charm store, it's not possible to determine the
+        OpenStack release codename and in that case it will be considered as ussuri.
         :return: OpenStackRelease object
         :rtype: OpenStackRelease
         """
-        try:
-            # get the OpenStack release from the channel track of the application.
-            os_track_release_channel = OpenStackRelease(self.channel.split("/", maxsplit=1)[0])
-        except ValueError:
-            logger.debug(
-                "The current channel of '%s' does not exist or is unexpectedly formatted",
-                self.name,
-            )
-            os_track_release_channel = self.current_os_release
-        return os_track_release_channel
+        if self.is_from_charm_store:
+            return OpenStackRelease("ussuri")
+
+        # applications with no workload version will call channel_codename without setting
+        # the channel before
+        if not hasattr(self, "channel"):
+            self.channel = self.status.charm_channel
```

Comment:
Why dis is not define at property, it really strange to set property value in getter for other property.

Reply:
`channel_codename` can be called before setting `channel` during `_populate_units`. If we set the channel before populating, auxiliary charms will try to call `current_os_release` without populating previously. I don't have a better solution than this. It's a chicken and egg problem.

Reply:
How that is possible? The `channel` property will be there before any post_innit is executed or not ?

Reply:
1. `_populate_units` is above channel setter
2. `_get_latest_os_version_by_workload_version` calls `channel_codename` if the charm is openstack channel based
3. `channel_codename` calls self.channel that it's not declared yet.

A possible  option is leaving units without an OpenStack version when is channel based

Reply:
We can ensure that `_populate_units` will set channel before calling `_get_latest_os_version_by_workload_version`. However you mention that `_populate_units` is above channel setter, but here you are checking if `not hasattr(self, "channel")` what means that you are checking if self has property channel, which should be true always or not?